File: PropertyExchange/Presentation/PropertyExchange.Presentation.UI/src/app/shared/models/tenant.model.ts
import { Property } from "./property.model";

export class TenantDetails{
    Id:number;
    PropertyID: string;
    TenantID: string;
    Name: string;
    Description: string;
    Email: string;
    PhoneNumber: string;
    Country: string;
    IsActive: boolean;
    TenancyType: string;
    CreatedDate: string;
    LastUpdatedDate: string;
    Property: Property;
    LeaseDetails: TenantLeaseDetails[];
    RentPaymentDetails: TenantRentPaymentDetailsVM[];
}

export class TenantLeaseDetails{
    Id:number;
    PropertyID: string;
    TenantID: string;
    LeaseAdvanceAmount: number;
    LeaseStartDate: string;
    LeaseEndDate: string;
    LeaseTenureInMonths: number;
    RentPerSft: number;
    TotalAreaInSft: number;
    EscalationInPercentage: number;
    EscalationTenure: number;
    CreatedDate: string;
    LastUpdatedDate: string;
    Tenant: TenantDetails;
}

export class TenantRentPaymentDetailsVM{
    Id:number;
    PropertyID: string;
    TenantID: string;
    ExpenseType: string;
    Description: string;
    ExpenseAmount: number;
    ExpenseIncurredOnDate: string;
    ExpenseMonth: number;
    ExpenseYear: number;
    CreatedDate: string;
    LastUpdatedDate: string;
    Tenant: TenantDetails;
}

File: PropertyExchange/Core/PropertyExchange.Core.Domain/Models/Tenant/TenantModel.cs
using PropertyExchange.Presentation.API.Models.Property;

namespace PropertyExchange.Presentation.API.Models.Tenant
{
    public class TenantModel
    {
        public int Id { get; set; }
        public required string PropertyID { get; set; }
        public required string TenantID { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public required string Email { get; set; }
        public required string PhoneNumber { get; set; }
        public required string Country { get; set; }
        public required bool IsActive { get; set; }
        public required string TenancyType { get; set; }
        public required string CreatedDate { get; set; }
        public string? LastUpdatedDate { get; set; }
        public List<TenantLeaseDetailsModel>? LeaseDetails { get; set; }
        public List<TenantRentPaymentDetailsModel>? RentPaymentDetails { get; set; } 
        public PropertyModel? Property { get; set; }

    }
}
